src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworksList.tsx

```diff
@@ -1,197 +1,65 @@
-    },
-    partner: {
-      name: "Pace Gallery",
-    },
-    price: "",
-  },
-  {
-    internalID: "63dacd25feb14f000df4d272",
-    title: "U.A. Walker, New York",
-    artist: {
-      name: "Hiroshi Sugimoto",
-    },
-    partner: {
-      name: "Obscura Gallery",
-    },
-    price: "$2,250 - 3,250",
-  },
-  {
-    internalID: "6351bf3f835f69000eefbe55",
-    title: "Obliteration after Modigliani",
-    artist: {
-      name: "Richard Learoyd",
-    },
-    partner: {
-      name: "Fraenkel Gallery",
-    },
-    price: "",
-  },
-  {
-    internalID: "5b16e95b8b0c1413a6cdf588",
-    title: "David Standing on Water No. 1, Sherwood Island, Connecticut",
-    artist: {
-      name: "Rodney Smith",
-    },
-    partner: {
-      name: "Gilman Contemporary",
-    },
-    price: "$15,000 - 45,000",
-  },
-  {
-    internalID: "63dadb64feb14f000bddaf3a",
-    title: "Kiton Man, The Knole Estate, Long Island, New York",
-    artist: {
-      name: "Rodney Smith",
-    },
-    partner: {
-      name: "Robert Klein Gallery",
-    },
-    price: "$10,000 - 40,000",
-  },
-  {
-    internalID: "58fb5c39cd530e4d3dbbbdb3",
-    title: "Kate Moss, Nepal, British Vogue",
-    artist: {
-      name: "Arthur Elgort",
-    },
-    partner: {
-      name: "Atlas Gallery",
-    },
-    price: "$12,000 - 30,000",
-  },
-]
+import { createFragmentContainer, graphql } from "react-relay"
+import { extractNodes } from "Utils/extractNodes"
+import { ArtworksList_me$data } from "__generated__/ArtworksList_me.graphql"
 
 interface ArtworksListProps {
   selectedIds: string[]
   onItemClick: (artworkID: string) => void
+  me: ArtworksList_me$data
 }
 
-export const ArtworksList: FC<ArtworksListProps> = ({
-  selectedIds,
+const ArtworksList: FC<ArtworksListProps> = ({
   onItemClick,
+  selectedIds,
+  me,
 }) => {
+  const handleItemClick = (artworkID: string) => {
+    onItemClick(artworkID)
+  }
+
+  const artworks = extractNodes(me?.collection?.artworksConnection)
+
   return (
     <GridColumns>
-      {TEMP_ARTWORKS.map(artwork => {
+      {artworks.map(artwork => {
         return (
           <Column span={[6, 4]}>
             <ArtworkItem
               item={artwork}
               selected={selectedIds.includes(artwork.internalID)}
-              onItemClick={onItemClick}
+              onItemClick={handleItemClick}
             />
           </Column>
         )
       })}
     </GridColumns>
   )
 }
+
+export const ArtworksListFragmentContainer = createFragmentContainer(
+  ArtworksList,
+  {
+    me: graphql`
+      fragment ArtworksList_me on Me
+        @argumentDefinitions(
+          sort: { type: CollectionArtworkSorts, defaultValue: POSITION_DESC }
+        ) {
+        collection(id: "saved-artwork") {
+          name
+
+          artworksConnection(first: 30, sort: $sort) {
+            edges {
+              node {
+                internalID
+                ...GridItem_artwork
+              }
+            }
+          }
```

Comment:
**suggestion**: query this data in the `AddArtworksModalContent` component and pass it via `artworks` prop. Something like this

```javascript
// src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/AddArtworksModalContent.tsx
const AddArtworksModalContentRefetchContainer = createRefetchContainer(
  AddArtworksModalContent,
  {
    me: graphql`
      fragment AddArtworksModalContent_me on Me
        @argumentDefinitions(
          sort: { type: CollectionArtworkSorts, defaultValue: POSITION_DESC }
        ) {
        collection(id: "saved-artwork") {
          artworksConnection(first: 30, sort: $sort) {
            totalCount
            ...ArtworksList_artworks
          }
        }
      }
    `,
  },
  ...
)
...
<ArtworksListFragmentContainer
  artworks={me.collection?.artworksConnection!}
  onItemClick={handleItemClick}
  selectedIds={selectedArtworkIds}
/>


// src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal/ArtworksList.tsx
export const ArtworksListFragmentContainer = createFragmentContainer(
  ArtworksList,
  {
    artworks: graphql`
      fragment ArtworksList_artworks on ArtworkConnection {
        edges {
          node {
            internalID
            ...GridItem_artwork
          }
        }
      }
    `,
  }
)
```